# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../rpm/python24.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -Nur rpm-4.1.orig/lib/rpmts.h rpm-4.1/lib/rpmts.h
--- rpm-4.1.orig/lib/rpmts.h	2002-08-23 21:35:49.000000000 +0200
+++ rpm-4.1/lib/rpmts.h	2005-10-09 14:40:22.000000000 +0300
@@ -10,7 +10,7 @@
 
 /*@-exportlocal@*/
 /*@unchecked@*/
-extern int _rpmts_debug;
+/*extern int _rpmts_debug;
 /*@unchecked@*/
 extern int _fps_debug;
 /*@=exportlocal@*/
diff -Nur rpm-4.1.orig/python/header-py.c rpm-4.1/python/header-py.c
--- rpm-4.1.orig/python/header-py.c	2002-08-08 01:19:10.000000000 +0200
+++ rpm-4.1/python/header-py.c	2005-10-09 13:33:35.000000000 +0300
@@ -2,9 +2,10 @@
  * \file python/header-py.c
  */
 
-#include "system.h"
-
 #include "Python.h"
+
+#undef _POSIX_C_SOURCE
+#include "system.h"
 #ifdef __LCLINT__
 #undef  PyObject_HEAD
 #define PyObject_HEAD   int _PyObjectHead;
@@ -429,7 +430,7 @@
 	while (extensions->name) {
 	    if (extensions->type == HEADER_EXT_TAG
 		&& !xstrcasecmp(extensions->name + 7, str)) {
-		(const struct headerSprintfExtension *) ext = extensions;
+		ext = extensions;
 	    }
 	    extensions++;
 	}
diff -Nur rpm-4.1.orig/python/rpmal-py.c rpm-4.1/python/rpmal-py.c
--- rpm-4.1.orig/python/rpmal-py.c	2002-07-14 23:24:28.000000000 +0200
+++ rpm-4.1/python/rpmal-py.c	2005-10-09 12:38:39.000000000 +0300
@@ -2,9 +2,10 @@
  * \file python/rpmal-py.c
  */
 
-#include "system.h"
-
 #include "Python.h"
+#undef _POSIX_C_SOURCE
+
+#include "system.h"
 #ifdef __LCLINT__
 #undef  PyObject_HEAD
 #define PyObject_HEAD   int _PyObjectHead;
diff -Nur rpm-4.1.orig/python/rpmdb-py.c rpm-4.1/python/rpmdb-py.c
--- rpm-4.1.orig/python/rpmdb-py.c	2002-07-25 19:03:12.000000000 +0200
+++ rpm-4.1/python/rpmdb-py.c	2005-10-09 13:49:23.000000000 +0300
@@ -2,9 +2,10 @@
  * \file python/rpmdb-py.c
  */
 
-#include "system.h"
-
 #include "Python.h"
+#undef _POSIX_C_SOURCE
+
+#include "system.h"
 #ifdef __LCLINT__
 #undef  PyObject_HEAD
 #define PyObject_HEAD   int _PyObjectHead;
diff -Nur rpm-4.1.orig/python/rpmds-py.c rpm-4.1/python/rpmds-py.c
--- rpm-4.1.orig/python/rpmds-py.c	2002-07-19 22:26:39.000000000 +0200
+++ rpm-4.1/python/rpmds-py.c	2005-10-09 13:49:52.000000000 +0300
@@ -2,9 +2,10 @@
  * \file python/rpmds-py.c
  */
 
-#include "system.h"
-
 #include "Python.h"
+#undef _POSIX_C_SOURCE
+
+#include "system.h"
 #ifdef __LCLINT__
 #undef  PyObject_HEAD
 #define PyObject_HEAD   int _PyObjectHead;
diff -Nur rpm-4.1.orig/python/rpmfd-py.c rpm-4.1/python/rpmfd-py.c
--- rpm-4.1.orig/python/rpmfd-py.c	2002-07-14 23:24:28.000000000 +0200
+++ rpm-4.1/python/rpmfd-py.c	2005-10-09 13:50:21.000000000 +0300
@@ -2,9 +2,10 @@
  * \file python/rpmfd-py.c
  */
 
-#include "system.h"
-
 #include "Python.h"
+#undef _POSIX_C_SOURCE
+
+#include "system.h"
 #ifdef __LCLINT__
 #undef  PyObject_HEAD
 #define PyObject_HEAD   int _PyObjectHead;
diff -Nur rpm-4.1.orig/python/rpmfi-py.c rpm-4.1/python/rpmfi-py.c
--- rpm-4.1.orig/python/rpmfi-py.c	2002-08-19 20:51:24.000000000 +0200
+++ rpm-4.1/python/rpmfi-py.c	2005-10-09 13:50:44.000000000 +0300
@@ -2,9 +2,10 @@
  * \file python/rpmfi-py.c
  */
 
-#include "system.h"
-
 #include "Python.h"
+#undef _POSIX_C_SOURCE
+
+#include "system.h"
 #ifdef __LCLINT__
 #undef  PyObject_HEAD
 #define PyObject_HEAD   int _PyObjectHead;
diff -Nur rpm-4.1.orig/python/rpmmi-py.c rpm-4.1/python/rpmmi-py.c
--- rpm-4.1.orig/python/rpmmi-py.c	2002-08-10 01:14:10.000000000 +0200
+++ rpm-4.1/python/rpmmi-py.c	2005-10-09 13:51:07.000000000 +0300
@@ -2,9 +2,10 @@
  * \file python/rpmmi-py.c
  */
 
-#include "system.h"
-
 #include "Python.h"
+#undef _POSIX_C_SOURCE
+
+#include "system.h"
 #ifdef __LCLINT__
 #undef  PyObject_HEAD
 #define PyObject_HEAD   int _PyObjectHead;
diff -Nur rpm-4.1.orig/python/rpmmodule.c rpm-4.1/python/rpmmodule.c
--- rpm-4.1.orig/python/rpmmodule.c	2002-08-13 18:37:52.000000000 +0200
+++ rpm-4.1/python/rpmmodule.c	2005-10-09 13:21:01.000000000 +0300
@@ -2,9 +2,10 @@
  * \file python/rpmmodule.c
  */
 
-#include "system.h"
-
 #include "Python.h"
+#undef _POSIX_C_SOURCE
+
+#include "system.h"
 #ifdef __LCLINT__
 #undef  PyObject_HEAD
 #define PyObject_HEAD   int _PyObjectHead;
@@ -422,7 +423,7 @@
 
     while (extensions->name) {
 	if (extensions->type == HEADER_EXT_TAG) {
-            (const struct headerSprintfExtension *) ext = extensions;
+            ext = extensions;
             PyDict_SetItemString(d, (char *) extensions->name, o=PyCObject_FromVoidPtr(ext, NULL));
 	    Py_DECREF(o);
             PyDict_SetItem(dict, tag, o=PyString_FromString(ext->name + 7));
diff -Nur rpm-4.1.orig/python/rpmrc-py.c rpm-4.1/python/rpmrc-py.c
--- rpm-4.1.orig/python/rpmrc-py.c	2002-08-05 23:46:52.000000000 +0200
+++ rpm-4.1/python/rpmrc-py.c	2005-10-09 13:41:44.000000000 +0300
@@ -2,9 +2,10 @@
  * \file python/rpmrc-py.c
  */
 
-#include "system.h"
-
 #include "Python.h"
+#undef _POSIX_C_SOURCE
+
+#include "system.h"
 #ifdef __LCLINT__
 #undef  PyObject_HEAD
 #define PyObject_HEAD   int _PyObjectHead;
diff -Nur rpm-4.1.orig/python/rpmte-py.c rpm-4.1/python/rpmte-py.c
--- rpm-4.1.orig/python/rpmte-py.c	2002-07-14 23:24:29.000000000 +0200
+++ rpm-4.1/python/rpmte-py.c	2005-10-09 13:48:27.000000000 +0300
@@ -2,9 +2,10 @@
  * \file python/rpmte-py.c
  */
 
-#include "system.h"
-
 #include "Python.h"
+#undef _POSIX_C_SOURCE
+
+#include "system.h"
 #ifdef __LCLINT__
 #undef  PyObject_HEAD
 #define PyObject_HEAD   int _PyObjectHead;
diff -Nur rpm-4.1.orig/python/rpmts-py.c rpm-4.1/python/rpmts-py.c
--- rpm-4.1.orig/python/rpmts-py.c	2002-09-03 14:26:09.000000000 +0200
+++ rpm-4.1/python/rpmts-py.c	2005-10-09 14:39:57.000000000 +0300
@@ -2,9 +2,10 @@
  * \file python/rpmts-py.c
  */
 
-#include "system.h"
-
 #include "Python.h"
+#undef _POSIX_C_SOURCE
+
+#include "system.h"
 #ifdef __LCLINT__
 #undef  PyObject_HEAD
 #define PyObject_HEAD   int _PyObjectHead;
